integration_test/gce/gce_testing.go

```diff
@@ -1229,32 +1243,80 @@ func attemptCreateInstance(ctx context.Context, logger *log.Logger, options VMOp
+func verifyVMCreation(ctx context.Context, logger *log.Logger, vm *VM) error {
+	if err := waitForStart(ctx, logger, vm); err != nil {
+		return err
 	}
-	newLabels, err := addFrameworkLabels(options.Labels)
-	if err != nil {
-		return nil, fmt.Errorf("attemptCreateInstance() could not construct valid labels: %v", err)
+
+	if os, err := getOS(ctx, logger, vm); err != nil {
+		return err
+	} else {
+		vm.OS = *os
 	}
 
-	imageFamilyScope := options.ImageFamilyScope
+	if IsSUSEVM(vm) {
+		// Set download.max_silent_tries to 5 (by default, it is commented out in
+		// the config file). This should help with issues like b/211003972.
+		if _, err := RunRemotely(ctx, logger, vm, "sudo sed -i -E 's/.*download.max_silent_tries.*/download.max_silent_tries = 5/g' /etc/zypp/zypp.conf"); err != nil {
+			return fmt.Errorf("attemptCreateInstance() failed to configure retries in zypp.conf: %v", err)
+		}
+	}
 
-	if imageFamilyScope == "" {
-		imageFamilyScope = "global"
+	if IsSLESVM(vm) {
+		if err := prepareSLES(ctx, logger, vm); err != nil {
+			return fmt.Errorf("%s: %v", prepareSLESMessage, err)
+		}
 	}
 
-	args := []string{
-		// "beta" is needed for --max-run-duration below.
-		"beta", "compute", "instances", "create", vm.Name,
-		"--project=" + vm.Project,
-		"--zone=" + vm.Zone,
-		"--machine-type=" + vm.MachineType,
-		"--image-family-scope=" + imageFamilyScope,
-		"--network=" + vm.Network,
-		"--format=json",
+	if IsSUSEVM(vm) {
+		// Set ZYPP_LOCK_TIMEOUT so tests that use zypper don't randomly fail
+		// because some background process happened to be using zypper at the same time.
+		if _, err := RunRemotely(ctx, logger, vm, `echo 'ZYPP_LOCK_TIMEOUT=300' | sudo tee -a /etc/environment`); err != nil {
+			return err
+		}
+	}
+
+	// Removing flaky rhel-7 repositories due to b/265341502
+	if isRHEL7SAPHA(vm.ImageSpec) {
+		if _, err := RunRemotely(ctx,
+			logger, vm, `sudo yum -y --disablerepo=rhui-rhel*-7-* install yum-utils && sudo yum-config-manager --disable "rhui-rhel*-7-*"`); err != nil {
+			return fmt.Errorf("disabling flaky repos failed: %w", err)
+		}
 	}
+
+	// Pre-installed jupyter services on DLVM images cause port conflicts for third-party apps.
+	// See b/347107292.
+	if IsDLVMImage(vm.ImageSpec) {
+		if _, err := RunRemotely(ctx, logger, vm, "sudo service jupyter stop || true"); err != nil {
+			return fmt.Errorf("attemptCreateInstance() failed to stop pre-installed jupyter service: %v", err)
+		}
+	}
+
+	// TODO(b/369656678): We see frequent errors like "Waiting for cache lock:
+	//   Could not get lock /var/lib/dpkg/lock-frontend", so add a generous timeout.
+	if IsDebianBased(vm.ImageSpec) {
+		if _, err := RunRemotely(ctx, logger, vm, "echo 'DPkg::Lock::Timeout=300' | sudo tee /etc/dpkg/dpkg.cfg.d/cache-lock-timeout.cfg"); err != nil {
+			return fmt.Errorf("setting increased dpkg cache lock timeout failed: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func addditionalCreateInstaceArgs(options VMOptions, vm *VM) ([]string, error) {
```

Comment:
Nit: I think you meant `additionalCreateInstanceArgs` (two typos)

Reply:
Fixed. Thank you!